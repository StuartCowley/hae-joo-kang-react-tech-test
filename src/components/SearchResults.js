import React from "react";
import PropTypes from "prop-types";
import "../styles/search-results.css";

const SearchResults = ({ results, isLoading }) => {
  if (!results.length && !isLoading) {
    return <h2>No matching images. Try again.</h2>
  } 
  return (
    <>
    {isLoading && <h2>Please enter search terms</h2>}
    <div className="search-results">
      {results.map((image, index) => (
        <img 
          src={image} 
          className="search-results__image" 
          alt={`${image}`} 
          key={index}
        />
      ))}
    </div>
    </>
  )
};

SearchResults.propTypes = {
  results: PropTypes.arrayOf(PropTypes.string).isRequired,
  isLoading: PropTypes.bool.isRequired
};

export default SearchResults;
