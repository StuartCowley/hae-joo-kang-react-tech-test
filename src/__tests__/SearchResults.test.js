import { render, screen } from "@testing-library/react";
import React from "react";
import renderer from "react-test-renderer";
import SearchResults from "../components/SearchResults";

describe("SearchResults", () => {
  let mockResults = [
      "https://testimage1",
      "https://testimage2",
      "https://testimage3",
  ];

  it("renders SearchResults correctly", () => {
    const rendered = renderer.create(<SearchResults results={mockResults}/>);
    expect(rendered).toMatchSnapshot();
  });

  it("renders correct number of results", () => {
    render(<SearchResults results={mockResults} />);

    expect(screen.getByAltText("https://testimage1")).toBeInTheDocument();
    expect(screen.getByAltText("https://testimage2")).toBeInTheDocument();
    expect(screen.getByAltText("https://testimage3")).toBeInTheDocument();
  });

  it("shows 'No results' message when the results is empty", () => {
    mockResults = [];

    render(<SearchResults results={mockResults} />);

    expect(screen.getByText("No results")).toBeInTheDocument();
  })
});