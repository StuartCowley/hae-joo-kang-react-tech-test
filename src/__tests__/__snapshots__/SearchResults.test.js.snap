// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResults renders SearchResults correctly 1`] = `
<div
  className="search-results"
>
  <img
    alt="https://testimage1"
    className="search-results__image"
    src="https://testimage1"
  />
  <img
    alt="https://testimage2"
    className="search-results__image"
    src="https://testimage2"
  />
  <img
    alt="https://testimage3"
    className="search-results__image"
    src="https://testimage3"
  />
</div>
`;
